dart/lib/src/sentry_client.dart

```diff
@@ -97,6 +114,14 @@ class SentryClient {
           .getSentryException(event.throwable, stackTrace: event.stackTrace);
 
       event = event.copyWith(exception: sentryException);
+    } else if (_options.attachStackTrace &&
+        event.stackTrace == null &&
+        event.exception == null) {
+      final frames = _stackTraceFactory.getStackFrames(StackTrace.current);
```

Comment:
you should use the given `stackTrace` if not null and only if null and `attachStacktrace` is enabled, you use `StackTrace.current`

Reply:
✅

Reply:
wrong condition IMO, the `else if` branch was right, but it's missing this:

if (stackTrace == null) stackTrace = StackTrace.current
and:
final frames = _stackTraceFactory.getStackFrames(stackTrace);

Reply:
✅